### PR TITLE
First draft of swarm syntax documentation

### PR DESCRIPTION
**Pull request by aheinz-fe**
I didn't document pivot, as we don't use it and I am less familiar with it. I may also have documented syntax for features in unmerged PR.

### CODE REVIEW

File: synapse/swarm/SYNTAX.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+Regular Expression
+
+    foo:bar~="^foo"
+
+## Extended Comparisons
+
+Value In Range
+range(x,y) is equivalent to x <= value < y
+
+    foo:bar*range(1,2)
```
Comment (invisig0th) [inline synapse/swarm/SYNTAX.md]: foo:bar*range=(1,13)

File: synapse/swarm/SYNTAX.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+    ... -foo:baz=10
+    ... -foo:baz="ten"
+
+## Join
+Supported comparisons: Basic
+
+Join all foo tufos whose foo:bar prop matches a baz:faz prop in our data set.
+
+    ... <foo:bar=baz:faz
```
Comment (invisig0th) [inline synapse/swarm/SYNTAX.md]: not a huge fan of < as join macro due to being confused with less than. liked & but it's now in use for logic... thoughts?

Comment (aheinz-fe) [inline synapse/swarm/SYNTAX.md]: Well, we actually use [ under the hood, but I map < to [ before it gets to swarm.

Comment (aheinz-fe) [inline synapse/swarm/SYNTAX.md]: Another option would be to look at && or other doubled characters.

**Comment by invisig0th** [comment]
too many changes to syntax and terminology to port this.  easier to start again :D

**Pull request closed without merging**